web3/encoding.nim

```diff
@@ -1,216 +1,178 @@
 import
-  std/[typetraits, strutils, macros, math]
+  std/macros,
+  stint, ./ethtypes
 
-import
-  stint, stew/byteutils, ./ethtypes
-
-type
-  EncodeResult* = tuple[dynamic: bool, data: string]
-
-func encode*[bits: static[int]](x: StUint[bits]): EncodeResult =
-  ## Encodes a `StUint` to a textual representation for use in the JsonRPC
-  ## `sendTransaction` call.
-  (dynamic: false, data: '0'.repeat((256 - bits) div 4) & x.dumpHex)
-
-func encode*[bits: static[int]](x: StInt[bits]): EncodeResult =
-  ## Encodes a `StInt` to a textual representation for use in the JsonRPC
-  ## `sendTransaction` call.
-  (dynamic: false,
-  data:
-    if x.isNegative:
-      'f'.repeat((256 - bits) div 4) & x.dumpHex
-    else:
-      '0'.repeat((256 - bits) div 4) & x.dumpHex
-  )
+func encode*[bits: static[int]](x: StUint[bits]): seq[byte] =
+  @(x.toByteArrayBE())
+
+func encode*[bits: static[int]](x: StInt[bits]): seq[byte] =
+  @(x.toByteArrayBE())
 
-func decode*(input: string, offset: int, to: var StUint): int =
-  let meaningfulLen = to.bits div 8 * 2
-  to = type(to).fromHex(input[offset .. offset + meaningfulLen - 1])
+func decode*(input: openarray[byte], baseOffset, offset: int, to: var StUint): int =
+  const meaningfulLen = to.bits div 8
+  let offset = offset + baseOffset
+  to = type(to).fromBytesBE(input[offset ..< offset + meaningfulLen])
```

Comment:
hm, these decode functions assume sufficient input data, or they will panic - it's not a new issue, but looks risky

Reply:
Just keeping to status quo here.